### PR TITLE
Create presubmit job for kubevirt/libguestfs-appliance

### PR DESCRIPTION
**Pull request by alicefr**
Create presubmit prow job for the respository [libguestfs-appliance](https://github.com/kubevirt/libguestfs-appliance)

### CODE REVIEW

File: github/ci/prow-deploy/files/jobs/kubevirt/libguestfs-appliance/libguestfs-appliance-presubmits.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    always_run: true
+    optional: false
+    decorate: true
+    decoration_config:
+      timeout: 3h
+    max_concurrency: 1
+    labels:
+      preset-dind-enabled: "true"
+      preset-docker-mirror: "true"
+    cluster: phx-prow
```
Comment (rmohr) [inline github/ci/prow-deploy/files/jobs/kubevirt/libguestfs-appliance/libguestfs-appliance-presubmits.yaml:13]: We don't need the bare-metal cluster. You can place here `cluster: ibm-prow-jobs`

File: github/ci/prow-deploy/files/jobs/kubevirt/libguestfs-appliance/libguestfs-appliance-presubmits.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        - "/usr/local/bin/runner.sh"
+        - "/bin/sh"
+        - "-c"
+        - "./create-libguestfs-appliance.sh"
+        # docker-in-docker needs privileged mode
+        securityContext:
+          privileged: true
+        resources:
+          requests:
+            memory: "8Gi"
```
Comment (rmohr) [inline github/ci/prow-deploy/files/jobs/kubevirt/libguestfs-appliance/libguestfs-appliance-presubmits.yaml:29]: 2 or 4 Gi are probably still more than enough.

File: github/ci/prow-deploy/files/jobs/kubevirt/libguestfs-appliance/libguestfs-appliance-presubmits.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    decorate: true
+    decoration_config:
+      timeout: 3h
+    max_concurrency: 1
+    labels:
+      preset-dind-enabled: "true"
+      preset-docker-mirror: "true"
+    cluster: phx-prow
+    spec:
+      nodeSelector:
```
Comment (rmohr) [inline github/ci/prow-deploy/files/jobs/kubevirt/libguestfs-appliance/libguestfs-appliance-presubmits.yaml:15]: This node selector is not needed. We only need it on the bare metal server.

**Review by rmohr — CHANGES_REQUESTED** [review]
A few simplifications are possible.

File: github/ci/prow-deploy/files/jobs/kubevirt/libguestfs-appliance/libguestfs-appliance-presubmits.yaml (hunk 1 above)
Comment (alicefr) [reply github/ci/prow-deploy/files/jobs/kubevirt/libguestfs-appliance/libguestfs-appliance-presubmits.yaml]: thanks, I was not sure which one to pick :)

**Comment by alicefr** [comment]
it cannot find the `main` branch but it still searches for `master`

**Comment by rmohr** [comment]
> it cannot find the `main` branch but it still searches for `master`

ah right. that is a rehearse plugin bug.

/hold cancel

**Comment by alicefr** [comment]
I listed `main` explicitly under branches. Not sure if this is the right way to do it

**Comment by rmohr** [comment]
> I listed `main` explicitly under branches. Not sure if this is the right way to do it

`main` must be set, but in addition  the rehearse plugin only supports `master` I think. Let's bring the job in and test with a real PR.

File: github/ci/prow-deploy/files/jobs/kubevirt/libguestfs-appliance/libguestfs-appliance-presubmits.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
+presubmits:
+  kubevirt/libguestfs-appliance:
+  - name: check-build-libguestfs-appliance
```
Comment (rmohr) [inline github/ci/prow-deploy/files/jobs/kubevirt/libguestfs-appliance/libguestfs-appliance-presubmits.yaml:3]: the recommended naming pattern would be `pull-libgiuestfs-appliance-build`

Comment (alicefr) [reply github/ci/prow-deploy/files/jobs/kubevirt/libguestfs-appliance/libguestfs-appliance-presubmits.yaml]: ok, I changed it.

**Pull request merged**